frontend/src/components/Menu/DropdownMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  Box,
+  Button,
+  Menu as ChakraMenu,
+  MenuButton as ChakraMenuButton,
+  MenuItem as ChakraMenuItem,
+  MenuItemProps,
+  MenuList as ChakraMenuList,
+  MenuListProps,
+  MenuProps,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+
+import { BxsChevronDown } from '~/assets/icons/BxsChevronDown'
+import { BxsChevronUp } from '~/assets/icons/BxsChevronUp'
+import {
+  DropdownMenuSize,
+  DropdownMenuVariant,
+} from '~/theme/components/DropdownMenu'
+
+export interface DropdownMenuButtonProps {
+  /**
+   * The variant of the dropdown menu.
+   */
+  variant: DropdownMenuVariant
+  /**
+   * Whether the menu has been opened
+   */
+  isActive: boolean
+  /**
+   * The size of the dropdown menu - stretched or not.
+   */
+  size: DropdownMenuSize
```

Comment:
should we check with pearly on how the component determines the sizing? it could be that on focus, it stretches to the width (some inputs do this) or on large screens, it automatically becomes stretched. 

if that's the case, it would make more sense to determine the sizing internally and not expose it.

Reply:
thanks! changed to dynamic sizing as per discussion on figma, and similar to button - two options are available (instead of determining it internally)